src/structures/Widget.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const Base = require('./Base');
+const Collection = require('../util/Collection');
+
+/**
+ * Represents a Widget.
+ */
+class Widget extends Base {
+  /**
+   * @param {Client} client - The instantiating client
+   * @param {Object} data - The raw data;
+   */
+  constructor(client, data) {
+    super(client);
+    this._patch(data);
+  }
+  /**
+   * Builds the widget with the provided data.
+   * @param {*} data The raw data of the widget
+   * @private
+   */
+  _patch(data) {
+    /**
+     * When a widget is disable, data will always send a code. When it's enable, data doesn't have a code.
+     */
+    if ('code' in data) {
```

Comment:
I don't really understand why you're creating a Widget class for a JSON error. I'd suggest throwing a `WIDGET_DISABLED` error or something instead of constructing this class at all.

Reply:
I don't know at all how to make it :(
But yeah it might be a good idea

Reply:
I think the rest of the constructor (in the `else` part) is fine. But if the Widget is disabled, check that back in the `fetchWidget` function and reject the Promise instead of returning a Widget